cognite_toolkit/_cdf_tk/prototypes/_packages/update_references.py

```diff
@@ -0,0 +1,338 @@
+        for name, cls in filtered_classes.items():
+            if cls._scope_name == "all":
+                scope_members["all"] = {}
+            else:
+                if "ids" in cls.__annotations__:
+                    scope_members[f"*{cls._scope_name}"] = {"*ids": "list[str]"}
+                else:
+                    scope_members[f"{cls._scope_name}"] = {}
+
+        expanded.append(
+            {
+                acl._capability_name: {
+                    "actions": action_members,
+                    "scope": scope_members
+                    if len(scope_members) == 1
+                    else {"*one of": {"*" + k: v for k, v in scope_members.items()}},
+                }
+            }
+        )
+    return expanded
+
+
+def expand_transformation_destinations() -> dict[str, Any]:
+    options: dict[str, Any] = {}
+    options["*type"] = []
+    for name, member in inspect.getmembers(TransformationDestination):
+        if isinstance(TransformationDestination.__dict__.get(name), staticmethod):
+            signature = inspect.signature(member)
+            if len(signature.parameters) == 0:
+                options["*type"].append(f'*"{name}"')
+            else:
+                options[f"*{to_camel_case(name)}"] = {k: f"*{v.annotation}" for k, v in signature.parameters.items()}
+    return {"*one of": options}
+
+
+def get_parameters(cls: Any) -> Any:
+    sign = inspect.signature(cls.__init__)
+
+    filtered_params = [param for name, param in sign.parameters.items() if name != "self" and name != "cognite_client"]
+    return sign.replace(parameters=filtered_params).parameters.values()
+
+
+def unpack_annotations(str_annotations: str) -> Any:
+    str_annotations = str_annotations.replace("Sequence[", "list[")
+    str_annotations = re.sub(r"Literal\[(.*?)\]", lambda match: match.group(1), str_annotations)
+
+    pattern = r"\|(?![^\[\]]*\])"
+    annotations = [a.strip() for a in re.split(pattern, str_annotations)]
+
+    if not any([a not in IGNORED_ANNOTATIONS for a in annotations]):
+        return str_annotations
+
+    cognite_classes = {name: obj for name, obj in inspect.getmembers(data_classes) if inspect.isclass(obj)}
+    cognite_classes.update({name: obj for name, obj in inspect.getmembers(dm_data_classes) if inspect.isclass(obj)})
+
+    resolved: dict[str, Any] = {}
+
+    for annotation in annotations:
+        if annotation.startswith("list["):
+            return [unpack_annotations(re.sub(r"list\[(.*?)\]", lambda match: match.group(1), annotation))]
+        else:
+            cls = cognite_classes.get(annotation)
+            if not cls:
+                continue
+            for param in get_parameters(cls):
+                resolved[to_camel_case(param.name)] = unpack_annotations(str(param.annotation))
+
+        return resolved
+
+    return str_annotations
+
+
+def expand_parameters(loader_cls: Any, cls: Any, optional: bool = False) -> dict[str, Any]:
```

Comment:
This feels like reinventing the wheel. I think the same can be achieved with the parameter module with some modifications. But less discuss whether it is worth the effort.

Reply:
As discussed IRL, you're right. This is a temporary solution.